GTBitsOfGood/southface
# hc/98-admin-edit-standard

## Editing + Deleting A Standard (Admin)

Issue Number(s): #98

What does this PR change and why?
This PR adds the edit view on clicking standards for admins. It allows you to change the fields of the standard in the card view. This is so admins can edit the standards quickly and easily from the main library view. It also allows them to delete standards.

### Checklist
- [x] Edit button is shown for admin users (just check user.isAdmin from useUser hook)
- [x] OnClick of edit, it renders the components as editable (look into the editable chakra UI component + talk to Alex)
- [x] All of the editing is done using [ReactFinalForm](https://final-form.org/react) again talk to Alex 
- [x] onClick of delete standard button, it deletes the entire standard (modify frontend and database)
- [x] all relevant "are you sure" modals are displayed
- [x] if fields are left empty, error states are shown (look into it with react final form)
- [x] all relevant backend database query/mutations are updated (database reflects changes)
- [ ] add image popup/uploading

### How to Test
- Go to http://localhost:3000/library/single-family/durability-and-moisture-management
- Sign out
- Click a standard
- See if there's an edit button, there shouldn't be one
- Log in as admin
- Click the standard again
- Click edit
- Change a bunch of fields (Criteria, tags, images, should all update)
- Click save
- Confirm
- Check that every major action has a confirm dialogue (Delete, edit)
- Check to see card is updated with the right info
- Click the standard again
- Check that delete works

### Notes
Add image functionality is there, just authentication is broken
There's no logic to check if a tag is valid or not and this is only a problem because for some reason you aren't supposed to be able to add new tags.
